frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/useFeatureToggleSwitch.tsx

```diff
@@ -215,7 +216,14 @@ export const useFeatureToggleSwitch: UseFeatureToggleSwitchType = (
     const modals = (
         <>
             <FeatureStrategyProdGuard {...prodGuardModalState} />
-            <EnableEnvironmentDialog {...enableEnvironmentDialogState} />
+            <ConditionallyRender
+                condition={enableEnvironmentDialogState.featureId.length !== 0}
```

Comment:
can you add a variable with intention revealing name. It may not be clear why featureId.length is the trigger here

Reply:
Yes, good